pulp/apis/gurobi_api.py

```diff
@@ -197,20 +293,24 @@ def buildSolverModel(self, lp):
                 lp.solverModel.setParam("LogFile", logPath)
 
             log.debug("add the variables to the problem")
+            lp.solverModel.update()
+            nvars = lp.solverModel.NumVars
             for var in lp.variables():
                 lowBound = var.lowBound
                 if lowBound is None:
-                    lowBound = -gurobipy.GRB.INFINITY
+                    lowBound = -gp.GRB.INFINITY
                 upBound = var.upBound
                 if upBound is None:
-                    upBound = gurobipy.GRB.INFINITY
+                    upBound = gp.GRB.INFINITY
                 obj = lp.objective.get(var, 0.0)
-                varType = gurobipy.GRB.CONTINUOUS
+                varType = gp.GRB.CONTINUOUS
                 if var.cat == constants.LpInteger and self.mip:
-                    varType = gurobipy.GRB.INTEGER
-                var.solverVar = lp.solverModel.addVar(
-                    lowBound, upBound, vtype=varType, obj=obj, name=var.name
-                )
+                    varType = gp.GRB.INTEGER
+                # only add variable once, ow new variable will be created.
```

Comment:
Added this check for https://github.com/coin-or/pulp/blob/a7684ac90d93dc3bca382d90841a9ab5f398b50f/pulp/tests/test_pulp.py#L660
Otherwise `x` is added twice and the test fails.